Stratis.Bitcoin/BlockPulling/LookaheadBlockPuller.cs

```diff
@@ -372,13 +372,16 @@ private void AskBlocks()
             if (this.location == null)
                 throw new InvalidOperationException("SetLocation should have been called");
 
-            if (this.lookaheadLocation == null && !this.Chain.Contains(this.location))
+            if ((this.lookaheadLocation != null) && (this.location != null) && (this.lookaheadLocation.Height < this.location.Height))
```

Comment:
Won't it be more readable to do 
```
if (this.lookaheadLocation?.Height < this.location?.Height)
        this.lookaheadLocation = this.location;
```

Reply:
Just commit if this comment does not apply.

Reply:
I'm still undecided about those question marks and double question marks. I think I tend to use them in cases like `loop?.Dispose()`, but in comparison it seems too much for me at the moment. Merged.